src/hdmf/utils.py

```diff
@@ -436,7 +423,7 @@ def dec(func):
             def func_call(*args, **kwargs):
                 self = args[0]
                 parsed = __parse_args(
-                            _copy.deepcopy(loc_val),
+                            loc_val,
```

Comment:
This is one of the key changes. Previously, this list of dictionaries (the docval specification/validator) was deepcopied on every call to a function with docval.